### PR TITLE
Add script to create a build drop from a sign job

### PR DESCRIPTION
**Pull request by cormacpayne**
## Comments

Fix for issue https://github.com/Azure/azure-powershell/issues/2929

This script will be used for Azure PowerShell releases.

It takes the build artifacts from a sign job and creates a new entry in the shared folder that will contain the msi, packages, scripts and WebPI installers from the PSReleaseDrop folder, and other xml files used for WebPI testing.

This script will also create and push a new branch to the given CodePlex WebPI fork containing the changes to the two xml files.

---

This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. For the categories that you did not check off, you can remove them from this body. Within each of the categories that you did select, make sure that you can check off **all** of the boxes. 

For information on cleaning up the commits in your pull request, click [here](../documentation/cleaning-up-commits.md).
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
- [x] [**MANDATORY** - Add/remove/edit test(s)](#tests)
### General
- [x] Title of the PR is clear and informative
- [x] There are a small number of commits that each have an informative message
- [x] If it applies, references the bug/issue that the PR fixes
- [x] All files have the Microsoft copyright header
- [x] Cmdlets refer to management libraries through nuget references - no dlls are checked in
- [x] The PR does not introduce breaking changes (unless a major version change occurs in the assembly and module)
### Tests
- [x] PR includes test coverage for the included changes
- [x] Tests must use xunit, and should either use Moq to mock management client calls, or use the scenario test framework
- [x] PowerShell scripts used in tests must not use hard-coded values for location
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for existing resources
- [x] Tests should not use App.config files for settings
- [x] Tests should use the built-in PowerShell functions for generating random names when unique names are necessary - this will store names in the test recording
- [x] Tests should use Start-Sleep to pause rather than Thread.Sleep

### CODE REVIEW

File: tools/BuildDrop.ps1 [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    # Check if we are looking at the DiscoveryHints for DH_WindowsAzurePowerShellVSFeed
+    # and if we have reached the end of the entry so we can add the new msi Product Code
+    if ($VSFeedSeen -and $content[$idx] -like "*</or>*")
+    {
+        $newContent[$idx] =     "      <discoveryHint>"
+        $newContent[$idx + 1] = "        <msiProductCode>$ProductCode</msiProductCode>"
+        $newContent[$idx + 2] = "      </discoveryHint>"        
+
+        # Change the buffer size to include the three lines just added
+        $buffer = 3
```
Comment (markcowl) [inline tools/BuildDrop.ps1]: Should we change $VSFeedSeen to false at this point?

File: tools/BuildDrop.ps1 [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    }
+
+    # Check if we are looking at the entry for DH_WindowsAzurePowerShellGet
+    if ($PSGetSeen -and $content[$idx] -like "*msiProductCode*")
+    {
+        $content[$idx] = "   <msiProductCode>$ProductCode</msiProductCode>"
+    }
```
Comment (markcowl) [inline tools/BuildDrop.ps1]: Shoudl we change $PSGetSeen to false at this point?

File: tools/BuildDrop.ps1 [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        {
+            $content[$idx] = "  <updated>$($ReleaseDate)T12:00:00Z</updated>"
+        }
+
+        if ($content[$idx] -like "*<trackingURL>*")
+        {
+            $content[$idx] = "        <trackingURL>http://www.microsoft.com/web/handlers/webpi.ashx?command=incrementproddownloadcount&amp;prodid=WindowsAzurePowershell&amp;version=$PSVersion&amp;prodlang=en</trackingURL>"
+        }
+    }
+
```
Comment (markcowl) [inline tools/BuildDrop.ps1]: Should have a condition here which ends the $PSGetSeen, to make this more robust over changed orderings in the file

File: tools/BuildDrop.ps1 (hunk 1 above)
Comment (cormacpayne) [inline tools/BuildDrop.ps1]: I thought about it, but didn't account for the case where the ordering of the file could change. That is a good idea. I will mark as false here.

File: tools/BuildDrop.ps1 (hunk 2 above)
Comment (cormacpayne) [inline tools/BuildDrop.ps1]: Same as comment for $VSFeedSeen.

File: tools/BuildDrop.ps1 (hunk 3 above)
Comment (cormacpayne) [inline tools/BuildDrop.ps1]: Same as comment for $VSFeedSeen.

**Pull request merged**